src/Servers/HttpSys/src/MessagePump.cs

```diff
@@ -275,4 +228,91 @@ public void Dispose()
 
         Listener.Dispose();
     }
+
+    private class AcceptLoop : IThreadPoolWorkItem
+    {
+        private readonly AsyncAcceptContext _asyncAcceptContext;
+        private readonly MessagePump _messagePump;
+        private readonly int _workerIndex;
+        private readonly bool _preferInlineScheduling;
+
+        public AcceptLoop(AsyncAcceptContext asyncAcceptContext,
+                          MessagePump messagePump,
+                          int workerIndex)
+        {
+            _asyncAcceptContext = asyncAcceptContext;
+            _messagePump = messagePump;
+            _workerIndex = workerIndex;
+            _preferInlineScheduling = _messagePump._options.UnsafePreferInlineScheduling;
+        }
+
+        public void Execute()
+        {
+            _ = ExecuteAsync();
+        }
+
+        private async Task ExecuteAsync()
+        {
+            while (!_messagePump.Stopping && _workerIndex <= _messagePump._maxAccepts)
```

Comment:
nit: move the `_maxAccepts` check to `ProcessRequestsWorker`. It should never be false. Tier 1 JIT can probably remove it, but we should just change the code.

Could also remove the `_workerIndex` field since it doesn't seem to really be used, unless it's for crash dumps.

Reply:
Agreed. I've been staring at this for a while. But we can remove `_acceptorCounts` too. It adds complexity for no reason.